File: modules/vescape-core/ios/config/RefloatConfigWriteVerifier.swift
import Foundation

enum RefloatConfigWriteVerification {
  case success
  case failure(String)
}

/// Confirms a config write took by comparing the patched bytes against the board readback, surfacing
/// the same failure vocabulary as Android.
///
/// @parity /modules/vescape-core/android/src/main/java/expo/modules/vescapecore/config/RefloatConfigWriteVerifier.kt
enum RefloatConfigWriteVerifier {
  static func verifyExactBytes(expected: [UInt8], actual: [UInt8]) -> RefloatConfigWriteVerification {
    if expected == actual { return .success }

    if expected.count != actual.count {
      return .failure(
        "Verification failed: expected \(expected.count) bytes, read back \(actual.count) bytes"
      )
    }

    let firstMismatch = expected.indices.first { expected[$0] != actual[$0] } ?? 0
    return .failure("Verification failed: first byte mismatch at offset \(firstMismatch)")
  }
}

File: modules/vescape-core/android/src/main/java/expo/modules/vescapecore/config/RefloatConfigWriteVerifier.kt
package expo.modules.vescapecore.config

internal sealed class RefloatConfigWriteVerification {
  object Success : RefloatConfigWriteVerification()
  data class Failure(val message: String) : RefloatConfigWriteVerification()
}

// @parity /modules/vescape-core/ios/config/RefloatConfigWriteVerifier.swift
internal object RefloatConfigWriteVerifier {
  fun verifyExactBytes(expected: ByteArray, actual: ByteArray): RefloatConfigWriteVerification {
    if (expected.contentEquals(actual)) return RefloatConfigWriteVerification.Success

    if (expected.size != actual.size) {
      return RefloatConfigWriteVerification.Failure(
        "Verification failed: expected ${expected.size} bytes, read back ${actual.size} bytes",
      )
    }

    val firstMismatch = expected.indices.first { expected[it] != actual[it] }
    return RefloatConfigWriteVerification.Failure(
      "Verification failed: first byte mismatch at offset $firstMismatch",
    )
  }
}
